server/test/unit/server/test_config.py

```diff
@@ -0,0 +1,223 @@
+                                         restore_config_attrs)
+from pulp.server import config
+
+# a fake config with a new section and one that overrides a default
+FAKE_CONFIG_1 = '''\
+[section1]
+key1 = value1
+
+[server]
+test_key = test_value
+server_name = foo
+'''
+
+# another fake config
+FAKE_CONFIG_2 = '''\
+[section1]
+key2 = value2
+
+[section2]
+key1 = value3
+'''
+
+# save the existing config to restore when we're done messing with it
+initial_config_files = config._config_files
+initial_config_object = config.config
+
+
+class ConfigFileMock(object):
+    # This is defined in the static scope of the class to deal with tracking
+    # all calls to open in a single test across multiple instances;
+    # use (mock_instance).load_counts.clear() to reset its state
+    load_counts = defaultdict(int)
+
+    # the available version of mock.mock_open doesn't support the readline method,
+    # so this is a custom mock designed just for tracking config file loads,
+    # supporting only the file interface that ConfigParsers actually use
+    def __init__(self, name, *args, **kwargs):
+        # we're mocking open, so open doesn't work here
+        self.real_file = os.fdopen(os.open(name, os.O_RDWR), *args, **kwargs)
+        self.load_counts[name] += 1
+
+    def readline(self):
+        return self.real_file.readline()
+
+    def close(self):
+        self.real_file.close()
+
+
+class TestDefaultConfigFiles(unittest.TestCase):
+    @mock.patch.object(__builtin__, 'open')
+    def test_config_not_loaded(self, mock_open):
+        # We mess with the default config files *a lot* in following tests,
+        # so we ought to make sure the defaults are sane before messing around
+
+        # reload the module to make sure any of the aforementioned messing
+        # is un-done before we start testing it
+        reload(config)
+
+        # open wasn't called during module load
+        self.assertEqual(mock_open.call_count, 0)
+
+        # _config_files is what we expect
+        self.assertEqual(config._config_files, ['/etc/pulp/server.conf'])
+
+        # config is the right type
+        self.assertIsInstance(config.config, config.LazyConfigParser)
+
+    def tearDown(self):
+        block_load_conf()
+
+
+class TestLoadConfFromFiles(unittest.TestCase):
+    def setUp(self):
+        restore_config_attrs()
+        self.addCleanup(override_config_attrs)
+
+        self.tmpdir = tempfile.mkdtemp()
+        self.addCleanup(partial(shutil.rmtree, self.tmpdir))
+
+        config_1_fh, self.config_1_name = tempfile.mkstemp(dir=self.tmpdir)
+        config_1 = os.fdopen(config_1_fh, 'w')
+        config_1.write(FAKE_CONFIG_1)
+        config_1.close()
+
+        config_2_fh, self.config_2_name = tempfile.mkstemp(dir=self.tmpdir)
+        config_2 = os.fdopen(config_2_fh, 'w')
+        config_2.write(FAKE_CONFIG_2)
+        config_2.close()
+
+        # teh spoofs (spooves?)
```

Comment:
I thought it would be funny to leave a serious comment on an obvious joke, but I guess not. ![sadface](http://i.imgur.com/LWMP63e.gif)